### DISCUSSION TITLE
onBlur - Question

### DISCUSSION THREAD

**Discussion by LoncaSorin** [nick-keller/react-datasheet-grid#119] [Q&A] [answered]
Hello @nick-keller,

I'm trying to make a call to the BE using onBlur and it's pretty hard with the existing information. Could you add "value" to the "CellWithId" object that onBlur is getting as a param? I know that this is usually how other libraries are doing that. What do you think? Or could you give me an alternative, maybe I've missed something? The only solution I found is to use multiple events (onChange, onBlur, and onFocus) to gather the information from them and store it on the state.

I've also tried to overwrite the onBlur method and add a new param like:

<DataSheetGrid
  ...
  onBlur={({cell}) => handleCellBlur(cell, units)} 
/>

- because if I'm able to pass the updated units, with the information from "cell", I'm able to make the patch call to the BE
- but the handleCellBlur is getting the old units (somehow, I don't understand why), not the updated array

Thanks!

**Comment by nick-keller** [comment, marked as answer]
Hello @LoncaSorin ,
You can have a look at [this article](https://react-datasheet-grid.netlify.app/docs/recipes/tracking-rows-changes), it should help you sync the backend with your changes,

**Reply by LoncaSorin** [reply]
Thanks!
